common/nglinks.py

```diff
@@ -0,0 +1,116 @@
+"""Neuroglancer link functions."""
+import json
+import urllib
+from typing import Optional, Union
+from collections import OrderedDict
+from dataclasses import dataclass
+from enum import Enum
+
+
+class LayerType(Enum):
+    Image = "image"
+    Segmentation = "segmentation"
+
+
+@dataclass
+class Layer:
+    cloudpath: str
+    type: LayerType
```

Comment:
What are all of the content types that we would use here? Do only affinity maps use shaders? I don't for synapses. Is it worth supporting vector fields for alignment?

Reply:
I remember kisuk "abused" shaders to view semantic labels, because chunkgraph created everything as image. But I suppose the proper solution is to force "image" type in the info file or neuroglancer. I used seuron to create the synful vector fields. 
I suppose we can use the fields to indicate input and output layers. Right now the chunkflow dag also takes the IMAGE_SHADER and OUTPUT_SHADER from the parameter file and apply it to the output layer. The segmentation dag uses IMAGE_SHADER and AFF_SHADER, not sure the best way to combine them.

Reply:
I think is best left as an extension to this PR. I haven't written anything that would use this, and I don't think I clearly understand what you're suggesting.

Reply:
OK